src/Npgsql/PreparedStatementManager.cs

```diff
@@ -61,7 +61,8 @@ internal PreparedStatementManager(NpgsqlConnector connector)
         if (BySql.TryGetValue(sql, out var pStatement))
         {
             Debug.Assert(pStatement.State != PreparedState.Unprepared);
-            if (pStatement.IsExplicit)
+            // If statement is invalidated, fall through below where we replace it with another
+            if (pStatement.IsExplicit && pStatement.State != PreparedState.Invalidated)
```

Comment:
Strictly speaking, changes in this file aren't required for the fix. But they do allow to instead just call `Prepare` again after error 0A000 instead of calling both `Unprepare` and `Prepare` together. All in all I'm ok with removing this part (or for example just not backporting this particular change).

Reply:
Seems fine to me